integration/auth/user_mgt_test.go

```diff
@@ -91,6 +92,219 @@ func TestGetNonExistingUser(t *testing.T) {
 	}
 }
 
+func TestGetUsers(t *testing.T) {
+	var testUser1 *auth.UserRecord
+	var testUser2 *auth.UserRecord
+	var testUser3 *auth.UserRecord
+
+	importUser1UID := randomUID()
+	importUser1 := (&auth.UserToImport{}).
+		UID(importUser1UID).
+		Email(randomEmail(importUser1UID)).
+		PhoneNumber(randomPhoneNumber()).
+		ProviderData([](*auth.UserProvider){
+			&auth.UserProvider{
+				ProviderID: "google.com",
+				UID:        "google_" + importUser1UID,
+			},
+		})
+
+	createTestUsers := func(t *testing.T) {
+		testUser1 = newUserWithParams(t)
+		testUser2 = newUserWithParams(t)
+		testUser3 = newUserWithParams(t)
+		importUser(t, importUser1UID, importUser1)
+	}
+
+	// Checks to see if the users list contain the given uids. Order is ignored.
+	//
+	// Behaviour is undefined if there are duplicate entries in either of the
+	// slices.
+	//
+	// This function is identical to the one in auth/user_mgt_test.go
+	sameUsers := func(users [](*auth.UserRecord), uids []string) bool {
+		if len(users) != len(uids) {
+			return false
+		}
+
+		sort.Slice(users, func(i, j int) bool {
+			return users[i].UID < users[j].UID
+		})
+		sort.Slice(uids, func(i, j int) bool {
+			return uids[i] < uids[j]
+		})
+
+		for i := range users {
+			if users[i].UID != uids[i] {
+				return false
+			}
+		}
+
+		return true
+	}
+
+	createTestUsers(t)
+	defer deleteUser(testUser1.UID)
+	defer deleteUser(testUser2.UID)
+	defer deleteUser(testUser3.UID)
+	defer deleteUser(importUser1UID)
+
+	userRecordsToUIDs := func(users [](*auth.UserRecord)) []string {
+		results := []string{}
+		for i := range users {
+			results = append(results, users[i].UID)
+		}
+		return results
+	}
+
+	t.Run("returns users by various identifier types in a single call", func(t *testing.T) {
```

Comment:
Lengthy test names/descriptions are not all that common in Go. Although I haven't found any concrete guidelines on the topic either. I'd say take a look at the tenant management integration tests, where we've used short names. But I'll leave it up to you to decide whether to use that pattern here or not.

Reply:
Those ones seem to be using the function name as the test name/description. (Which makes sense since in those cases, the SDK is effectively just calling the RPC. We're doing a little extra processing here.) I can't apply that directly, since these tests all use the same function name, but lets see if I can add some terseness.